x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/cat/RestCatTrainedModelsAction.java

```diff
@@ -0,0 +1,262 @@
+
+public class RestCatTrainedModelsAction extends AbstractCatAction {
+
+    private static final Logger logger = LogManager.getLogger(RestCatTrainedModelsAction.class);
+
+    public RestCatTrainedModelsAction(RestController controller) {
+        controller.registerHandler(GET, "_cat/ml/trained_models/{" + TrainedModelConfig.MODEL_ID.getPreferredName() + "}", this);
+        controller.registerHandler(GET, "_cat/ml/trained_models", this);
+    }
+
+    @Override
+    public String getName() {
+        return "cat_ml_get_trained_models_action";
+    }
+
+    @Override
+    protected RestChannelConsumer doCatRequest(RestRequest restRequest, NodeClient client) {
+        String modelId = restRequest.param(TrainedModelConfig.MODEL_ID.getPreferredName());
+        if (Strings.isNullOrEmpty(modelId)) {
+            modelId = MetaData.ALL;
+        }
+        GetTrainedModelsStatsAction.Request statsRequest = new GetTrainedModelsStatsAction.Request(modelId);
+        GetTrainedModelsAction.Request modelsAction = new GetTrainedModelsAction.Request(modelId, false, null);
+        if (restRequest.hasParam(PageParams.FROM.getPreferredName()) || restRequest.hasParam(PageParams.SIZE.getPreferredName())) {
+            statsRequest.setPageParams(new PageParams(restRequest.paramAsInt(PageParams.FROM.getPreferredName(), PageParams.DEFAULT_FROM),
+                restRequest.paramAsInt(PageParams.SIZE.getPreferredName(), PageParams.DEFAULT_SIZE)));
+            modelsAction.setPageParams(new PageParams(restRequest.paramAsInt(PageParams.FROM.getPreferredName(), PageParams.DEFAULT_FROM),
+                restRequest.paramAsInt(PageParams.SIZE.getPreferredName(), PageParams.DEFAULT_SIZE)));
+        }
+        statsRequest.setAllowNoResources(restRequest.paramAsBoolean(ALLOW_NO_MATCH.getPreferredName(),
+            statsRequest.isAllowNoResources()));
+        modelsAction.setAllowNoResources(restRequest.paramAsBoolean(ALLOW_NO_MATCH.getPreferredName(),
+            statsRequest.isAllowNoResources()));
+
+        return channel -> {
+            final ActionListener<Table> listener = ActionListener.notifyOnce(new RestResponseListener<>(channel) {
+                @Override
+                public RestResponse buildResponse(final Table table) throws Exception {
+                    return RestTable.buildResponse(table, channel);
+                }
+            });
+
+            client.execute(GetTrainedModelsAction.INSTANCE, modelsAction, ActionListener.wrap(
+                trainedModels -> {
+                    final List<TrainedModelConfig> trainedModelConfigs = trainedModels.getResources().results();
+
+                    Set<String> potentialAnalyticsIds = new HashSet<>();
+                    // Analytics Configs are created by the XPackUser
+                    trainedModelConfigs.stream()
+                        .filter(c -> XPackUser.NAME.equals(c.getCreatedBy()))
```

Comment:
I think we want a better way of differentiating user models and DFA models in the future. Maybe a reserved tag for DFA models

Reply:
@davidkyle possibly. Users cannot set the `created_by`, and `XPackUser.NAME` is a reserved name. 

But I see your point for models we provide as a resource. Those weren't created by a DFA.